icu4c/source/i18n/unicode/numberformatter.h

```diff
@@ -1517,9 +1517,10 @@ class U_I18N_API Padder : public UMemory {
     friend class impl::GeneratorHelpers;
 };
 
-// Do not enclose entire MacroProps with #ifndef U_HIDE_INTERNAL_API, needed for a protected field
+// Do not enclose entire MacroProps with #ifndef U_HIDE_INTERNAL_API, needed for a protected field.
+// U_I18N_API because intltest uses it.
 /** @internal */
-struct U_I18N_API MacroProps : public UMemory {
+struct U_I18N_API_CLASS MacroProps : public UMemory {
```

Comment:
I don't understand why this struct would ever need to be exported: It's fully defined in the header file, from which the compiler gets all of it for use in any compilation unit. What is there that could possibly be in the DLL that would need to be exported from there? My assumption was that it had simply been declared U_I18N_API because of overly eager copy-pasting, not because it had any reason to be. Am I missing something here?

Reply:
Compilers may or may not inline, and this is used from outside the i18n library.